### PR TITLE
PP-446 Add receipt email personalisation

### PR DESCRIPTION
**Pull request by mrlumbu**
## WHAT

_A brief description of the pull request:_
In order to inform users that a payment has been successful
we need to push charge and service data to our notification
provider (currently GOV.UK Notify) so that our templated email
is completed correctly. Currently our template requires six
fields:

`serviceReference, description, amount, date, serviceName, customParagraph`

A couple of small notes:
- We format the date in the connector (not ideal but currently unavoidable).
  All dates are in London time.
- The amount is sent through in form xx.yy - the pound sign is in the template
  itself as it seems notifies java client does not correctly encode it
  ## HOW
  
  _Steps to test or reproduce:_

Currently tricky to test locally. You need to insert following into your connector.env 
NOTIFY_EMAIL_ENABLED=true
NOTIFY_EMAIL_TEMPLATE=<email-template-id>
NOTIFY_SERVICE_ID=<pay-notify-service-id>
NOTIFY_SECRET=<pay-notify-secret>

where those ids can be provided by other channels. Local testing situation should improve soon.
## WHO

_Who should review this pull request:_
- [x] Developers
- [] WebOps

### CODE REVIEW

**Comment by rauligs** [comment]
rebuild

File: src/main/java/uk/gov/pay/connector/service/UserNotificationService.java [hunk 1]
```diff
@@ -74,4 +80,28 @@ private void readEmailConfig(ConnectorConfiguration configuration) {
             throw new RuntimeException("config property 'emailTemplateId' is missing or not set, which needs to point to the email template on the notify");
         }
     }
+
+    private Map<String, String> buildEmailPersonalisationFromCharge(ChargeEntity charge) {
+        String serviceName = charge.getGatewayAccount().getServiceName() != null ?
+                charge.getGatewayAccount().getServiceName() :
+                "";
```
Comment (rahul-vinayak) [inline src/main/java/uk/gov/pay/connector/service/UserNotificationService.java]: minor: easier and cleaner to use StringUtils.defaultString()

File: src/main/java/uk/gov/pay/connector/service/UserNotificationService.java [hunk 2]
```diff
@@ -74,4 +80,28 @@ private void readEmailConfig(ConnectorConfiguration configuration) {
...
+        String templateBody = charge.getGatewayAccount().getEmailNotification().getTemplateBody() != null ?
+                charge.getGatewayAccount().getEmailNotification().getTemplateBody() :
+                "";
+
```
Comment (rahul-vinayak) [inline src/main/java/uk/gov/pay/connector/service/UserNotificationService.java]: and here, StringUtils.defaultString()

Comment (mrlumbu) [inline src/main/java/uk/gov/pay/connector/service/UserNotificationService.java]: ah, good to know, thanks

File: src/test/java/uk/gov/pay/connector/unit/service/UserNotificationServiceTest.java [hunk 3]
```diff
@@ -109,7 +150,22 @@ public void testEmailSendWhenEmailsNotifyDisabled() throws Exception {
...
+            this.expectedParameters = expectedParameters;
+        }
+
+        public boolean matches(Object actualEmailRequest) {
+            return ((EmailRequest) actualEmailRequest)
+                    .getPersonalisation()
+                    .asMap()
+                    .equals(expectedParameters);
+        }
+    }
```
Comment (rahul-vinayak) [inline src/test/java/uk/gov/pay/connector/unit/service/UserNotificationServiceTest.java]: 👍

**Pull request merged**